src/modules/identity/RadixRemoteIdentity.ts

```diff
@@ -0,0 +1,226 @@
+import RadixECIES from '../crypto/RadixECIES'
+import RadixIdentity from './RadixIdentity'
+import RadixAccount from '../account/RadixAccount'
+import RadixSignature from '../atom/RadixSignature'
+import RadixSerializer from '../serializer/RadixSerializer'
+
+import { RadixAtom, RadixKeyPair } from '../RadixAtomModel'
+
+import WebSocket from 'isomorphic-ws'
+
+export default class RadixRemoteIdentity extends RadixIdentity {
+    private token: string
+    private remoteUrl: string
+    private socket: WebSocket
+
+    private constructor(readonly keyPair: RadixKeyPair, token: string, remoteUrl: string) {
+        super(keyPair)
+
+        this.token = token
+        this.remoteUrl = remoteUrl
+    }
+
+    /**
+     * Checks if a given web socket connection is still alive, otherwise creates a new one
+     * 
+     * @param _socket - WebSocket connection
+     * @returns A WebSocket connection
+     */
+    private getSocketConnection(_socket: WebSocket): WebSocket {
+        if (!_socket || _socket.readyState === _socket.CLOSING || _socket.readyState === _socket.CLOSED) {
+            _socket = new WebSocket(this.remoteUrl)
+        }
+        return _socket
+    }
+
+    /**
+     * Signs an atom with the wallet using the remote identity
+     * 
+     * @param atom - The atom to be signed
+     * @returns A promise with the signed atom
+     */
+    public signAtom(atom: RadixAtom): Promise<RadixAtom> {
+        return new Promise<RadixAtom>((resolve, reject) => {
+            const socket = this.getSocketConnection(this.socket)
+
+            socket.onopen = () => {
```

Comment:
Does this code work? It seems to me that if the socket connection is already open, the code in the `onopen` method would not get triggered, or am I missing something?

Also, what happens if I make a `signAtom` request, and before I get a response back I make a `decryptECIESPayload` request? The `onmessage` function would get reassigned and the response would be dropped.

Is there a reason you don't want to use the `rpc-websockets` library we already use in RadixNodeConnection? It deals with all these problems https://github.com/elpheria/rpc-websockets

Reply:
To be honest I don't have a reason, I just took the old-school approach. You're right in trying to add the least amount of libraries so I'll replace `isomorphic-ws` with `rpc-websockets`, thanks for the advice by the way.

The code works, but regarding the `onmessage` function it makes sense what you're saying, though that could be solved by having a `handlingResponse` method that deals with all the possible responses (sounds like a lot of work).